### ISSUE TITLE
Execute test class in multiproject

### ISSUE THREAD

**Issue by paulobrizolara** [eclipse/buildship#409]
I am porting a large gradle project to a multiproject configuration and face a problem with test execution from eclipse.

While in command line i can test a single class with something like:

         ./gradlew subproject:test --tests "com.sample.myproject.MyTestClass"

this way it only runs 'MyTestClass' tests.

However, when running from eclipse (right click in a class file or definition and select 'Run As > Gradle Test'),  apparently it will run something like:

         ./gradlew test --tests "com.sample.myproject.MyTestClass"

i.e., it will run the test task for all subprojects.

In my case, i can notice it because a dependent task (that shows the test report) is executed once for each subproject.
While i am trying to workaround this, i think it could benefit everyone if only the submodule task is executed (the build would be more efficient).

Thanks.

Obs.: if matters i am using TestNG, Gradle 3.3 and Buildship version 2.0.0.v20170111-1029 with Eclipse Neon.2 Release (4.6.2).

**Comment by oehme** [comment]
Indeed, the `TestLauncher` API that Buildship delegates to could be smarter about this. 

> In my case, i can notice it because a dependent task (that shows the test report) is executed once for each subproject.

Could you elaborate what this task does?

**Comment by loetifuss** [comment]
Similar problem here with using Gradle test suites (more related to https://github.com/eclipse-buildship/buildship/issues/489). When running a simple unit test Buildship tries to execute all available test tasks of all suites including integration tests (which causes servers to be fired up etc.). This makes it unusable to run in a local IDE. With IntelliJ IDEA only the relevant test task (i.e. 'test") is executed.
Attached is an example project reproducing the issue. Try to run "CalculatorTest.java" as Gradle Test in Buildship.

[test-fixtures.zip](https://github.com/user-attachments/files/27960247/test-fixtures.zip)

Are there any known workarounds for this problem?
